utils/scripts/tag.js

```diff
@@ -0,0 +1,214 @@
+
+  const readFile = util.promisify(fs.readFile);
+  const INSERTION_SLUG = '<!-- insert-new-changelog-here -->';
+  const changelogPath = resolve(__dirname, '..', '..', 'CHANGELOG.md');
+  const data = await readFile(changelogPath, 'utf8');
+
+  if (data.includes(INSERTION_SLUG)) {
+    const writeFile = util.promisify(fs.writeFile);
+    const changes = (await readFile(path, 'utf8')).split('\n\n#### Committers')[0];
+
+    retVal = data.replace(INSERTION_SLUG, `${INSERTION_SLUG}\n${changes}`);
+    await writeFile(changelogPath, retVal);
+    await execa('git', [
+      'commit',
+      '-m',
+      `chore(changelog): add ${tag} [skip ci]`,
+      '--no-verify',
+      '--quiet',
+      changelogPath
+    ]);
+  } else {
+    throw new Error(`Missing "${INSERTION_SLUG}" in CHANGELOG.md`);
+  }
+
+  return retVal;
+};
+
+/**
+ * Push a GitHub release.
+ *
+ * @param {String} tag The tag for the new release.
+ * @param {String} markdown Markdown content.
+ * @param {Ora} spinner Terminal spinner.
+ *
+ * @returns The draft release URL.
+ */
+const release = async (tag, markdown, spinner) => {
+  info('Creating release...', spinner);
+  await execa('git', ['push', '--follow-tags', '--no-verify', '--atomic', 'origin', 'master']);
+
+  const url = await garden.githubRelease({ tag, body: markdown, spinner });
+
+  spinner.succeed(
+    `Success.\n🎗 Approve the draft release – ${url} – upon notification of a successful ${tag} publish to NPM.`
+  );
+};
+
+/**
+ * Rollback version tag and changelog updates.
+ *
+ * @param {String} tag The tag to rollback.
+ * @param {Ora} spinner Terminal spinner.
+ */
+const rollback = async (tag, spinner) => {
+  info(`Rolling back ${tag} changes...`, spinner);
+  await execa('git', ['tag', '-d', tag]);
+  // Undo the `version` and the `changelog` commits.
+  await execa('git', ['reset', '--hard', 'HEAD~2']);
+};
+
+/**
+ * Sync local master branch and tags with GitHub remote.
+ *
+ * @param {String} master Name for the master branch being used for tagging.
+ * @param {Ora} spinner Terminal spinner.
+ */
+const sync = async (master, spinner) => {
+  const branch = await garden.githubBranch(undefined /* path */, spinner);
+
+  if (branch === master) {
+    info('Syncing with remote...', spinner);
+    await execa('git', ['pull']);
+    await execa('git', ['fetch', '--tags', '--prune', '--prune-tags']);
+    spinner.stop();
+    await execa('yarn', ['install'], { stdout: process.stdout });
+  } else {
+    throw new Error(`Switch to the ${master} branch`);
+  }
+};
+
+/**
+ * Bump and publish package versions.
+ *
+ * @param {String} bump Semantic or explicit versioning bump.
+ * @param {String} preid Prerelease ID as defined by https://semver.org/#spec-item-9.
+ * @param {String} master Name for the master branch being used for tagging.
+ * @param {Ora} spinner Terminal spinner.
+ *
+ * @returns New version tag.
+ */
+const version = async (bump, preid, master, spinner) => {
```

Comment:
> use conventional commits logic to guess the right version bump

Could you clarify how this works? Does the `version` function look for the strings "fix" "feat" "BREAKING CHANGE" "!" from a set of commits from the last release to determine the right version bump?

Reply:
via [`lerna version`](https://github.com/lerna/lerna/tree/master/commands/version)